src/slash-commands/aoc-leaderboard/index.ts

```diff
@@ -0,0 +1,105 @@
+import type { AocLeaderboard } from '@prisma/client';
+import { differenceInMinutes } from 'date-fns';
+import { format } from 'date-fns';
+import { type ChatInputCommandInteraction, SlashCommandBuilder } from 'discord.js';
+import { Result } from 'oxide.ts';
+import { DAY_MONTH_YEAR_HOUR_MINUTE_FORMAT } from '../../utils/date';
+import { logger } from '../../utils/logger';
+import type { SlashCommand } from '../builder';
+import { fetchAndSaveLeaderboard, getAocSettings, getSavedLeaderboard } from './utils';
+
+export const DEFAULT_LEADERBOARD = 10;
+
+const data = new SlashCommandBuilder()
+  .setName('aoc-leaderboard')
+  .setDescription("Fetch advent of code leaderboard. Will get current year, or the previous one if it isn't Dec yet.");
+
+export function getAocYear(): number {
+  const DECEMBER = 11;
```

Comment:
December = 11 / 12?

Reply:
JS uses `monthIndex` instead of actual month, 0-11, 0 = Jan, 11 = Dec.

Reply:
TIL